app/controllers/products_controller.rb

```diff
@@ -41,7 +41,7 @@ def create
     @new_method = @type == :usr ? 'git' : params[:new_method]
 
     respond_to do |format|
-      if @product.save
+      if @product.save(context: @product.create_context)
```

Comment:
`Product#stage`, the method you split up, can be removed as it is never called. The code you call it is the use of `Product.stage`. They are two separate methods. They are both in the same file.